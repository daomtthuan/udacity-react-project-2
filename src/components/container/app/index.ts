import AppContainer from './_.view';

export default AppContainer;
