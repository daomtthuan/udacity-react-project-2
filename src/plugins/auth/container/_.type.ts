import { PropsWithChildren } from 'react';

export type AuthContainerProps = PropsWithChildren;
