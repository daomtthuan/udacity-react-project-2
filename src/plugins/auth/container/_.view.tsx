import { Location, Navigate, useLocation } from 'react-router-dom';
import { useAppSelector } from '~plugins/store';

import { AuthContainerProps } from './_.type';

const pathConfig = {
  signIn: '/',
  home: '/home',

  guest: ['/', '/sign-up'],
};

export default function AuthContainer({ children }: Readonly<AuthContainerProps>) {
  const location: Location<string | null> = useLocation();

  const authUser = useAppSelector((state) => state.auth.user);

  // Not authenticated
  if (!authUser) {
    // Location is not guest routes
    if (!pathConfig.guest.includes(location.pathname)) {
      return <Navigate to={pathConfig.signIn} state={location.pathname} />;
    }

    return children;
  }

  // Authenticated
  else {
    // Location has state
    if (location.state) {
      return <Navigate to={location.state} state={null} />;
    }

    // Location is guest routes
    if (pathConfig.guest.includes(location.pathname)) {
      return <Navigate to={pathConfig.home} state={null} />;
    }

    return children;
  }
}
