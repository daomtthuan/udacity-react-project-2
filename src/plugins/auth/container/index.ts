import AuthContainer from './_.view';

export default AuthContainer;
