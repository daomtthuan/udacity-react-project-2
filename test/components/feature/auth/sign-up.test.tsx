import AuthSignUp from '~components/feature/auth/sign-up';
import { TestContainer } from '~test/mocks/container';

import { render } from '@testing-library/react';

describe('AuthSignUp renders correctly', () => {
  test('AuthSignUp render as expected', () => {
    const component = render(
      <TestContainer>
        <AuthSignUp />
      </TestContainer>,
    );

    expect(component.baseElement).toMatchSnapshot();
  });
});
