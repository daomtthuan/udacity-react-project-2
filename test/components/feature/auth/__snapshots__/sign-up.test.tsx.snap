// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthSignUp renders correctly AuthSignUp render as expected 1`] = `
<body>
  <div>
    <form>
      <div
        class="my-3"
      >
        <label
          class="form-label"
          for="input-user"
        >
          User
        </label>
        <input
          autocomplete="username"
          class="form-control"
          id="input-user"
          name="userId"
          type="text"
        />
      </div>
      <div
        class="my-3"
      >
        <label
          class="form-label"
          for="input-name"
        >
          Name
        </label>
        <input
          autocomplete="name"
          class="form-control"
          id="input-name"
          name="name"
          type="text"
        />
      </div>
      <div
        class="my-3"
      >
        <label
          class="form-label"
          for="input-password"
        >
          Password
        </label>
        <input
          autocomplete="new-password"
          class="form-control"
          id="input-password"
          name="password"
          type="password"
        />
      </div>
      <div
        class="my-3"
      >
        <label
          class="form-label"
          for="input-repeat-password"
        >
          Re-enter password
        </label>
        <input
          autocomplete="off"
          class="form-control"
          id="input-repeat-password"
          name="repeatPassword"
          type="password"
        />
      </div>
      <div
        class="mt-3 mb-2 d-flex align-items-center justify-content-end gap-2"
      >
        <a
          class="btn btn-outline-primary"
          href="/"
        >
          Back to Sign In
        </a>
        <button
          class="btn btn-primary"
          type="submit"
        >
          Sign Up
        </button>
      </div>
    </form>
  </div>
</body>
`;
